src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -243,6 +226,17 @@ void BtreeLBAManager::complete_transaction(
     pin_set.add_pin(get_pin(*e));
   }
 
+  std::vector<CachedExtentRef> to_clear;
+  to_clear.reserve(t.get_retired_set().size());
+  for (auto &e: t.get_retired_set()) {
+    if (e->is_logical() || is_lba_node(*e))
+      to_clear.push_back(e);
+  }
+  // need to call check_parent from leaf->parent
+  std::sort(
+    to_clear.begin(), to_clear.end(),
+    [](auto &l, auto &r) { return get_depth(*l) < get_depth(*r); });
+
```

Comment:
I'm wondering if we also need to make `pin_set.check_parent(pin)` atomic with `LogicalCachedExtent::retire_pin()`.